### ISSUE TITLE
Investigate adding ctags support 

### ISSUE THREAD

**Issue by Naatan** [Komodo/KomodoEdit#1629]
This could potentially hook into the new "Symbol Browser" functionality.

I have almost no familiarity with ctags, and still have a hard time wrapping my head around it. Any input/advice on how this could work would be appreciated.

My current impression is that ctags are a way of essentially navigating project symbols in a very explicit manner, please correct me if I'm wrong and if anyone wants to ELI5 what CTAGS are exactly that would be much appreciated.

Pinging @agroszer as he is the author of https://github.com/agroszer/koctags and undoubtedly knows more about this than I do.

**Comment by mitchell-as** [comment]
@Naatan ctags simply scans source files in a directory root, looking for function, class, namespace, etc. definitions and groups them all into a single "tags" file. It is formatted in such a way that you can "jump" a particular symbol. However, symbols are single words, not full names. For example, here are some sample symbols from Komodo's source tree:

```
Apply   src/chrome/komodo/content/project/customMenuProperties.js       /^function Apply() {$/;"        f
Apply   src/chrome/komodo/content/project/macroProperties.js    /^function Apply() {$/;"        f
Apply   src/chrome/komodo/content/project/simplePartProperties.js       /^function Apply() {$/;"        f
Apply   src/chrome/komodo/content/project/snippetProperties.js  /^function Apply() {$/;"        f
Apply   src/chrome/komodo/content/project/tutorialProperties.js /^function Apply(){$/;" f
Apply   src/chrome/komodo/content/run/commandproperties.js      /^function Apply()$/;"  f
```

We could hook ctags into our "symbol browser" functionality, but it wouldn't really be adding much more than we already have.

**Comment by Naatan** [comment]
Hmm I see. So is there any reason why someone would be wanting ctags in Komodo, aside from maybe that they are unaware of our new "symbol browser"?

**Comment by mitchell-as** [comment]
ctags has fantastic support for C and C++. (Perhaps some other languages that we do not support via codeintel.) Other than that, I don't see a need for them.

**Comment by Naatan** [comment]
Alright then, closing this in favour of the symbol browser.

For those reading this that are interested: Open Command (Ctrl+Shift+O) and navigate to the "Sections" entry.
